mobile-browser-based-version/src/helpers/training/training_manager.js

```diff
@@ -48,39 +49,44 @@ export class TrainingManager {
    * Train the task's model either alone or in a distributed fashion depending on the user's choice.
    * @param {Object} dataset the dataset to train on
    * @param {Boolean} distributedTraining train in a distributed fashion
+   * @param {personalizationType} personalizationTypeChosen the type of personalization chosen for the model.
    */
-  async trainModel(dataset, distributedTraining) {
+  async trainModel(
+    dataset,
+    distributedTraining,
+    personalizationTypeChosen = modelDefinition.personalizationType.NONE
+  ) {
     this.data = dataset.Xtrain;
     this.labels = dataset.ytrain;
 
-    /**
-     * If IndexedDB is turned on and the working model exists, then load the
-     * existing model from IndexedDB. Otherwise, create a fresh new one.
-     */
-    const modelParams = [
-      this.task.taskID,
-      this.task.trainingInformation.modelID,
-    ];
-    if (this.useIndexedDB && (await getWorkingModelMetadata(...modelParams))) {
-      this.model = await getWorkingModel(...modelParams);
-    } else {
-      this.model = await this.task.createModel();
+    switch (personalizationTypeChosen) {
+      case modelDefinition.personalizationType.NONE:
+        this.model = new modelDefinition.Model(this.task, this.useIndexedDB);
+        break;
+
+      case modelDefinition.personalizationType.INTEROPERABILITY:
+        this.model = new modelDefinition.InteroperabilityModel(
+          this.task,
+          this.useIndexedDB
+        );
+        break;
```

Comment:
It would be better to have a function modelDefinition, an additional constructor, that takes in also personalizationType. and then returns the appropriate object.

Reply:
Yep that sounds good having a function out of the classes that return the right model ! good idea